clang/include/clang/Basic/TargetInfo.h

```diff
@@ -216,7 +216,7 @@ enum OpenCLTypeKind : uint8_t {
 ///
 class TargetInfo : public TransferrableTargetInfo,
                    public RefCountedBase<TargetInfo> {
-  std::shared_ptr<TargetOptions> TargetOpts;
+  TargetOptions *TargetOpts;
```

Comment:
Upstream breaks internal APIs all the time. This one in particular seems pretty safe: the `TargetInfo` constructor now takes `TargetOptions &` instead of `const std::shared_ptr<TargetOptions> &`, which will result in a compilation error with an understandable diagnostic. The expected lifetime of the reference is documented, so I don't see this being an issue.